coldfront/core/project/models.py

```diff
@@ -243,6 +246,69 @@ def __str__(self):
     def natural_key(self):
         return (self.title,) + self.pi.natural_key()
 
+    def add_user(self, user, role_choice, signal_sender=None):
+        """
+        Adds a user to the project.
+
+        If a ProjectUser already exists, its role will be set to "Active" and its role updated.
+        Otherwise, creates a new ProjectUser.
+
+        Params:
+            user (User): User to add.
+            role_choice (ProjetUserRoleChoice): Role to give the project user.
+            signal_sender (str): Sender for the `project_activate_user` signal.
+        """
+        user_status_obj = ProjectUserStatusChoice.objects.get(name="Active")
+
+        project_user, _created = self.projectuser_set.update_or_create(
+            user=user,
+            defaults={
+                "status": user_status_obj,
+                "role": role_choice,
+            },
+        )
```

Comment:
```suggestion
        project_user, _created = self.projectuser_set.update_or_create(
            user=user,
            role=role_choice,
            status=user_status_obj
        )
```

I don't think these should be defaults.

The original logic looks like this:
```py
    project_user_obj = project_obj.projectuser_set.get(user=user_obj)
    project_user_obj.role = role_choice
    project_user_obj.status = project_user_active_status_choice
    project_user_obj.save()
```

So it is able to change the `role` and `status` of a project that already exists.